src/widgets/connector.ts

```diff
@@ -378,17 +378,45 @@ export class ConnectorView extends DOMWidgetView {
 
         const connectionTemplate = this.connectionsTemplates[key];
 
+        const form: { [key: string]: any } = {};
+
+        // get previous selected connection
+        const previousSelect = sessionStorage.getItem("selectConnection");
+
+        if (previousSelect) {
+          const prevConnectionTemplate = this.connectionsTemplates[previousSelect];
+          const { fields } = prevConnectionTemplate;
+          fields.forEach((field: { id: string; default?: string }) => {
+            const id = field.id;
+            const defaultValue = field.hasOwnProperty("default")
+              ? field["default"]
+              : "";
+            const formField = <HTMLSelectElement>this.el.querySelector(`#${id}`);
+            if (formField && formField.value != defaultValue) {
+              form[id] = formField.value;
+            }
+          });
+        }
```

Comment:
add some comments here, I think you're reading the previous form and storing it?

Reply:
[fixed](https://github.com/ploomber/jupysql-plugin/pull/75/files#diff-245ccc293a87304bf06e6d14a672d745b47a2cc1951a41def9f729b25ee669e1R387-R390)